inspirehep/modules/records/receivers.py

```diff
@@ -32,6 +30,28 @@
 from inspirehep.utils.date import create_valid_date
 from inspirehep.dojson.utils import get_recid_from_ref, classify_field
 
+from inspirehep.dojson.utils import get_recid_from_ref
+
+from inspirehep.utils.date import create_valid_date
+
+from invenio_indexer.signals import before_record_index
+
+import six
+
+from .signals import after_record_enhanced
+
+
+@before_record_index.connect
+def enhance_record(recid, json, *args, **kwargs):
```

Comment:
Please note that first argument should be `sender` which points to the connected Flask application.